### DISCUSSION TITLE
Can the O3D303 camera acquire ifm3d::buffer_id::GRAYSCALE_IMAGE and ifm3d::buffer_id::JPEG_IMAGE data?

### DISCUSSION THREAD

**Discussion by 520liangqiujuan** [ifm/ifm3d#460] [Q&A]
Hi!
Can the O3D303 camera acquire ifm3d::buffer_id::GRAYSCALE_IMAGE and ifm3d::buffer_id::JPEG_IMAGE data?
run to `auto grayscale = frame.get()->GetBuffer(ifm3d::buffer_id::GRAYSCALE_IMAGE); `
The following error message is reported
`Exception caught: Lib: A buffer with the requested buffer_id is not available.: buffer_id: 104`
Is it that the camera does not support the output of this kind of image?

**Comment by dekhanra** [comment]
Hello @520liangqiujuan,

The requested buffer_id s are not available for the O3D3xx devices and there is no 2D Imager on O3D3xx devices. So, JPEG Image is not available.
